app/src/main/java/org/luxoft/sdl_core/BleSupport/BluetoothHandler.java

```diff
@@ -0,0 +1,163 @@
+package org.luxoft.sdl_core;
+
+import android.bluetooth.BluetoothGattCharacteristic;
+import android.bluetooth.le.ScanResult;
+import android.content.Context;
+import android.os.Handler;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+import static android.bluetooth.BluetoothGattCharacteristic.PROPERTY_WRITE;
+
+class BluetoothHandler {
+    public BluetoothCentralManager central;
+    private static BluetoothHandler instance = null;
+    private BluetoothPeripheral mPeripheral = null;
+    private final Context context;
+    private final Handler handler = new Handler();
+
+    // Testing-only service with ability to notify and write
+    private static final UUID SDL_TESTER_SERVICE_UUID = UUID
+            .fromString("00001101-0000-1000-8000-00805f9b34fb");
+
+    // Testing-only characteristic with permissions to write
+    private static final UUID MOBILE_RESPONSE_CHARACTERISTIC = UUID
+            .fromString("00001103-0000-1000-8000-00805f9b34fb");
+
+    // Testing-only characteristic for notifications
+    private static final UUID MOBILE_REQUEST_CHARACTERISTIC = UUID
+            .fromString("00001102-0000-1000-8000-00805f9b34fb");
+
+
+    public static synchronized BluetoothHandler getInstance(Context context) {
+        if (instance == null) {
+            instance = new BluetoothHandler(context);
+        }
+        return instance;
+    }
+
+    private final BluetoothPeripheralCallback peripheralCallback = new BluetoothPeripheralCallback() {
+        @Override
+        public void onServicesDiscovered(BluetoothPeripheral peripheral) {
+            // Request a higher MTU, iOS always asks for 185
+            peripheral.requestMtu(185);
```

Comment:
@ychernysheva please move to private static final variable in the beginning of class

Reply:
Done in e8ca88e58483ac32df3c23ffc3c31eadcd4cab87